### PR TITLE
Parametrized NNO

### PR DESCRIPTION
**Pull request by Reijix**
I added parametrized natural numbers objects (definition 2.2 https://ncatlab.org/nlab/show/natural+numbers+object#withparams).

This entails:
- Adding `Categories.Object.NaturalNumbers.Parametrized`
- Extending `Categories.Object.NaturalNumbers.Properties.F-Algebras` with the algebras corresponding to PNNOs
- Adding `Categories.Object.NaturalNumbers.Properties.Parametrized` (not sure about the naming) to show that in CCCs `NNO` yields a `ParametrizedNNO`

Furthermore I needed two lemmas that I added to `Categories.Category.BinaryProducts`, `Categories.Object.Exponential` and `Categories.Category.CartesianClosed`, these are:
- `∘swap-inj : f ∘ swap ≈ g ∘ swap → f ≈ g` (plus `swap∘-inj`)
- `λ-inj : λg f ≈ λg g → f ≈ g`

I thought they are convenient to have so I added them to the respective files, but I can also remove them and use them locally.

### CODE REVIEW

File: src/Categories/Object/Exponential.agda [hunk 1]
```diff
@@ -47,6 +47,9 @@ record Exponential (A B : Obj) : Set (o ⊔ ℓ ⊔ e) where
              f ≈ g → λg X×A f ≈ λg X×A g
   λ-cong X×A {f = f} {g = g} f≡g = λ-unique X×A (β X×A ○ f≡g)
 
+  λ-inj : ∀ {X : Obj} (X×A : Product X A) {f g} → λg X×A f ≈ λg X×A g → f ≈ g
+  λ-inj X×A {f = f} {g = g} eq = (⟺ (β X×A)) ○ (∘-resp-≈ʳ ([ X×A ⇒ product  ]×-cong₂ eq (Equiv.refl))) ○ β X×A
```
Comment (JacquesCarette) [inline src/Categories/Object/Exponential.agda:51]: This proof is just complex enough that I think spelling it out equationally would be a good idea.

File: src/Categories/Object/NaturalNumbers/Properties/Parametrized.agda [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+open HomReasoning
+open Equiv
+
+open Terminal terminal
+
+NNO×CCC⇒PNNO : NNO → ParametrizedNNO
+NNO×CCC⇒PNNO nno = record 
+  { N = N 
+  ; isParametrizedNNO = record
```
Comment (JacquesCarette) [inline src/Categories/Object/NaturalNumbers/Properties/Parametrized.agda:35]: Could you please extract these large inline proofs to be in a where clause below the record? This makes downstream usage more comfortable and more efficient.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

File: src/Categories/Object/Exponential.agda (hunk 1 above)
Comment (Reijix) [reply src/Categories/Object/Exponential.agda]: Agreed, adjusted it.

File: src/Categories/Object/NaturalNumbers/Properties/Parametrized.agda (hunk 2 above)
Comment (Reijix) [reply src/Categories/Object/NaturalNumbers/Properties/Parametrized.agda:35]: Sure, done!

**Review by JacquesCarette — APPROVED** [review]
Sorry for the slowness. My term is ridiculously busy. Wonderful stuff, thank you!

**Pull request merged**